### ISSUE TITLE
Fix screen focus issues in Travis & Jenkins

### ISSUE THREAD

**Issue by aptmac** [rh-jmc-team/jmc-qa#16]
Xvfb allows for the ui tests to be ran by starting a virtual display server, however there seems to be a problem focusing on the JMC application window, or at least interacting with the UI components.

**Comment by aptmac** [comment]
Errors: 

```
Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 20.705 sec <<< FAILURE! - in org.openjdk.jmc.browser.uitest.jvmcompat.ConnectionMessageTest
org.openjdk.jmc.browser.uitest.jvmcompat.ConnectionMessageTest  Time elapsed: 20.704 sec  <<< ERROR!
org.jemmy.TimeoutExpiredException: State 'Waiting for 1 org.eclipse.swt.widgets.Shell controls to be found adhering to Text = JDK Mission Control' has not been reached in 10000 milliseconds

org.openjdk.jmc.browser.uitest.jvmcompat.ConnectionMessageTest  Time elapsed: 20.705 sec  <<< ERROR!
org.jemmy.TimeoutExpiredException: State 'Waiting for 1 org.eclipse.swt.widgets.Shell controls to be found adhering to Text = JDK Mission Control' has not been reached in 10000 milliseconds

Running org.openjdk.jmc.browser.uitest.jvmcompat.LocalConnectionNamingTest
Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 20.208 sec <<< FAILURE! - in org.openjdk.jmc.browser.uitest.jvmcompat.LocalConnectionNamingTest
org.openjdk.jmc.browser.uitest.jvmcompat.LocalConnectionNamingTest  Time elapsed: 20.208 sec  <<< ERROR!
org.jemmy.TimeoutExpiredException: State 'Waiting for 1 org.eclipse.swt.widgets.Shell controls to be found adhering to Text = JDK Mission Control' has not been reached in 10000 milliseconds

org.openjdk.jmc.browser.uitest.jvmcompat.LocalConnectionNamingTest  Time elapsed: 20.208 sec  <<< ERROR!
org.jemmy.TimeoutExpiredException: State 'Waiting for 1 org.eclipse.swt.widgets.Shell controls to be found adhering to Text = JDK Mission Control' has not been reached in 10000 milliseconds

Running org.openjdk.jmc.browser.uitest.CustomConnectionsTest
Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 20.162 sec <<< FAILURE! - in org.openjdk.jmc.browser.uitest.CustomConnectionsTest
org.openjdk.jmc.browser.uitest.CustomConnectionsTest  Time elapsed: 20.161 sec  <<< ERROR!
org.jemmy.TimeoutExpiredException: State 'Waiting for 1 org.eclipse.swt.widgets.Shell controls to be found adhering to Text = JDK Mission Control' has not been reached in 10000 milliseconds

org.openjdk.jmc.browser.uitest.CustomConnectionsTest  Time elapsed: 20.162 sec  <<< ERROR!
org.jemmy.TimeoutExpiredException: State 'Waiting for 1 org.eclipse.swt.widgets.Shell controls to be found adhering to Text = JDK Mission Control' has not been reached in 10000 milliseconds

Running org.openjdk.jmc.browser.uitest.ConnectionExportImportTest
Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 20.264 sec <<< FAILURE! - in org.openjdk.jmc.browser.uitest.ConnectionExportImportTest
org.openjdk.jmc.browser.uitest.ConnectionExportImportTest  Time elapsed: 20.264 sec  <<< ERROR!
org.jemmy.TimeoutExpiredException: State 'Waiting for 1 org.eclipse.swt.widgets.Shell controls to be found adhering to Text = JDK Mission Control' has not been reached in 10000 milliseconds

org.openjdk.jmc.browser.uitest.ConnectionExportImportTest  Time elapsed: 20.264 sec  <<< ERROR!
org.jemmy.TimeoutExpiredException: State 'Waiting for 1 org.eclipse.swt.widgets.Shell controls to be found adhering to Text = JDK Mission Control' has not been reached in 10000 milliseconds


Results :

Tests in error: 
org.openjdk.jmc.browser.uitest.ConnectionExportImportTest.org.openjdk.jmc.browser.uitest.ConnectionExportImportTest
  Run 1: ConnectionExportImportTest>MCJemmyTestBase.access$1:83->MCJemmyTestBase.initialize:89 Â» TimeoutExpired
  Run 2: ConnectionExportImportTest.org.openjdk.jmc.browser.uitest.ConnectionExportImportTest Â» TimeoutExpired

org.openjdk.jmc.browser.uitest.CustomConnectionsTest.org.openjdk.jmc.browser.uitest.CustomConnectionsTest
  Run 1: CustomConnectionsTest>MCJemmyTestBase.access$1:83->MCJemmyTestBase.initialize:89 Â» TimeoutExpired
  Run 2: CustomConnectionsTest.org.openjdk.jmc.browser.uitest.CustomConnectionsTest Â» TimeoutExpired

org.openjdk.jmc.browser.uitest.jvmcompat.ConnectionMessageTest.org.openjdk.jmc.browser.uitest.jvmcompat.ConnectionMessageTest
  Run 1: ConnectionMessageTest>MCJemmyTestBase.access$1:83->MCJemmyTestBase.initialize:89 Â» TimeoutExpired
  Run 2: ConnectionMessageTest.org.openjdk.jmc.browser.uitest.jvmcompat.ConnectionMessageTest Â» TimeoutExpired

org.openjdk.jmc.browser.uitest.jvmcompat.LocalConnectionNamingTest.org.openjdk.jmc.browser.uitest.jvmcompat.LocalConnectionNamingTest
  Run 1: LocalConnectionNamingTest>MCJemmyTestBase.access$1:83->MCJemmyTestBase.initialize:89 Â» TimeoutExpired
  Run 2: LocalConnectionNamingTest.org.openjdk.jmc.browser.uitest.jvmcompat.LocalConnectionNamingTest Â» TimeoutExpired

```

Source: https://api.travis-ci.org/v3/job/464431695/log.txt

**Comment by aptmac** [comment]
Perhaps get the tests scripts to run with xvfb in preparation for CI environments.

**Comment by aptmac** [comment]
I have the uitests starting to work on my local machine running a virtual display, but Travis isn't responding the same.

What was happening is the visual display was being created, but the RCP application was initialized with no dimensions. After looking into this from running JMC as an RCP application through Eclipse, I noticed that for the visual display to actually show the application I had to either click on the splash screen, or click on the 1x1 pixel window that was rendered, and then JMC would run as usual.

After further digging, I'm wondering what the best approach is. It seems that the running of the application and the running of JMC are handled separately. If I inserted some code somewhere into JMC (perhaps a logger), it wouldn't be executed unless I clicked the splash screen or selected the app. It looks like the RCP application code sets up the environment (life cycle, etc.) for running JMC, but then just waits for window focus before actually running jmc.
